### PR TITLE
fix: batching console log save calls in redux store

### LINKED ISSUE #16792: [Bug]: Performance hit with storing multiple logs together
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Currently, the console log feature stores each log one by one. So for 1000s of logs, there are a lot of redundant updates that go to the subscribed selectors which can be reduced to batched updates.

### Steps To Reproduce

Perf issue seen with console logs in a for loop.

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by ayushpahwa**
## Description

Currently, the console log feature stores each log one by one. So for 1000s of logs, there are a lot of redundant updates that go to the subscribed selectors which can be reduced to batched updates. This PR updates the redux store save call from one by one to storing an array directly

Fixes #16792 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/utils/AppsmithConsole.ts [hunk 1]
```diff
@@ -32,18 +33,8 @@ function getTimeStamp() {
...
-  category = LOG_CATEGORY.USER_GENERATED,
-) {
-  log({
-    ...ev,
-    severity,
-    timestamp,
-    category,
-  });
+function addLogs(logs: Log[]) {
+  dispatchAction(debuggerLog(logs));
```
Comment (SatishGandham) [inline app/client/src/utils/AppsmithConsole.ts:37]: Where is this action handled?

**Comment by SatishGandham** [comment]
@ayushpahwa , can you confirm if it reduced the number of debugger log actions.

File: app/client/src/utils/AppsmithConsole.ts (hunk 1 above)
Comment (ayushpahwa) [reply app/client/src/utils/AppsmithConsole.ts:37]: image
It is handled in the debuggerReducer.ts file

**Comment by ayushpahwa** [comment]
> @ayushpahwa , can you confirm if it reduced the number of debugger log actions.
Yes @SatishGandham, there is a significant improvement.

Current release:
image

This PR:
image

**Review by SatishGandham — APPROVED** [review]

**Comment by ayushpahwa** [comment]
Adding test plan approved label since it is a code change that affects perf. The functionality is covered with existing tests.

**Pull request merged**